### ISSUE TITLE
Question About Laser Source [mmclab]

### ISSUE THREAD

**Issue by koenigjaeger** [fangq/mmc#20]
I am trying to model a ring illumination system and from what I can see you have the config options of cfg.detpos, cfg.srctype and cfg.srcpos. 

**cfg.srctype** determines the type of beam you are using
**cfg.detpos** specifies the detector. What exactly does that mean though? I am working with photoacoustics, so I am mainly concerned with exciting my tissue sample with light not detecting it (for initial simulations at least). So does this config refer to something in diffuse optical imaging? Or measuring the power/energy of the beam?
**cfg.srcpos** seems to be what I am looking for, but it is only a 1x3 vector and if I want to mimic ring illumination I would need to define ~360 sources. Is there a way to do this currently with cfg.srcpos?

*Also if there is a better place to discuss question about the software not related to bugs/issues let me know.

**Comment by fangq** [comment]
regarding cfg.detpos, if you get an empty detphoton array, likely because 1) nphoton is too small so that no photon arrives at the detector, or 2) your detector position is not exactly on the mesh surface. see

http://mcx.space/wiki/index.cgi?Doc/FAQ#My_simulation_created_an_empty_history_file_why_is_that

to verify, you may plot the mesh and detpos and see if it is on the surface, try

`mcxpreview(cfg)
`

mcxpreview.m can be found in mcxlab (not mmclab) but is compatible with mmclab (transparency looks a big strange in octave).

https://github.com/fangq/mcx/blob/master/mcxlab/mcxpreview.m

for cfg.srcpos, there is no built-in ring-shaped pencil beam array (mcxlab has pencilarray for NxM grid, but not ring). You will have to write a loop to run 360 separate simulations per position and add the final fluence (or concatenate detphoton array). if the simulation time is significant, the overhead of splitting an array source to separate simulations is not going to be big.

for discussions, we have the mmc-uses mailing list, which has over 90 mmc users currently subscribing.

https://groups.google.com/forum/?hl=en#!forum/mmc-users

if the question is generic and can potentially helpful for other users, using the mailing list is preferred.

**Comment by koenigjaeger** [comment]
@fangq Well I just have one more clarification so cfg.detpos is acting like some kind of photon sensor? Is it suppose to be like a power meter? because when you run mmc(...) it returns the fluence/deposition, which makes be wonder what extra purpose the detector has?

**Comment by fangq** [comment]
yes, it acts like a photon detector and save individual photon's parth pathlengths (ppl) info, as well as other info, when they exits the domain from the vicinity of a detector (determined by the radius of the detector).

the partial pathlength and how to use it is explained in the below paper by David back in 2002

https://www.osapublishing.org/oe/abstract.cfm?uri=OPEX-10-3-159

partial path length has a symbol of L_i,j in Eq. 2, and is saved in the so called "history file". That is equivalent to the .mch file (h=history) in mcx.

**Comment by koenigjaeger** [comment]
Ah I see. Thanks for your help!
